### DISCUSSION TITLE
GeeekPi Fan Hat Genmon Communication Issue

### DISCUSSION THREAD

**Discussion by ssmoss** [jgyates/genmon#1142] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

Distributor ID: Debian
Description: Debian GNU/Linux 11 (bullseye)
Release: 11
Codename: bullseye

Linux 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux

Genmon 1.19.03

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

I have a Pi 3B+ with the GeeekPi Fan Hat and finally the PintSize.Me Generator HAT (connected in that order).

I ran the follow commands to setup the fan and enabled the onboard display:

```
sudo raspi-config
Navigate to `4 Performance Options` -> `P4 Fan` -> `Yes` -> `14` -> `60` -> `OK` -> `OK` -> `Finish` -> `YES`.
Raspberry Pi will reboot and take effect.
```

```
sudo raspi-config
Navigate to `Interface Options` -> `I2C` -> `Enable` -> `YES`.
```

After rebooting I would get the following notice in the logs:
```
Selected Controller: generac_evo_nexus
GenMon Loaded for site: xxx using python 3.9: VEnv: False
WARNING: Initialization not complete after 30 seconds, possible communication failure. Check cabling.
```

If ran the same commands above only selecting disable able instead everything started working again. Looked though the [gengpio.conf](https://github.com/jgyates/genmon/blob/master/conf/gengpio.conf) and [gengpioin.conf](https://github.com/jgyates/genmon/blob/master/conf/gengpioin.conf) and didn't see any references to PIN 14.

Is there some reason either of these commands could cause and issue with operation?

**Comment by skipfire** [comment, marked as answer]
14 is serial transmit, if the fan hat requires that then it won't be compatible with genmon through the typical uart config.  You could use USB to connect to the generator.

**Reply by ssmoss** [reply]
> 14 is serial transmit, if the fan hat requires that then it won't be compatible with genmon through the typical uart config. You could use USB to connect to the generator.

A that would be it and PIN 14 is only required for speed modulation which isn't necessary. Thanks for the clarification.
